api/src/main/java/org/apache/iceberg/io/CloseableIterable.java

```diff
@@ -117,6 +118,14 @@ public O next() {
     };
   }
 
+  static <E> CloseableIterable<E> combine(Iterable<? extends Iterable<E>> iterable,
+                                          ExecutorService executorService,
+                                          int parallelism) {
+    Preconditions.checkNotNull(executorService, "ExecutorService shouldn't be null");
```

Comment:
The error message should be "Invalid executor service: null".

Reply:
You may want to allow null if the parallelism is 1. That will make the calling code much easier.